datadog_checks_dev/datadog_checks/dev/docker.py

```diff
@@ -225,8 +233,7 @@ def set_up_with_retry():
         down=tear_down,
         on_error=on_error,
         sleep=sleep,
-        endpoints=endpoints,
-        conditions=docker_conditions,
+        conditions=docker_conditions if attempts is None else None,
```

Comment:
At first glance the need for this makes the approach look convoluted. I wonder if we shouldn't instead add an `attempts` parameter to `environment_run` and handle the retries directly on that function? That's my gut feeling before diving deeper.